Assumptions for work depth analysis

hodelcl opened the pull request:
This PR enables the user to enter assumptions for the symbols before calling the work depth analysis.

It uses a VS code `InputBox` to handle the input. Validity of the input is checked by a regex. If the check passes, the assumptions are forwarded to the work depth analysis.

The corresponding PR to the dace repo can be found [here](https://github.com/spcl/dace/pull/1363).

phschaad commented on src/components/dace_interface.ts:
```diff
@@ -909,20 +939,36 @@ implements vscode.WebviewViewProvider {
-                        this.genericErrorHandler(error.message, error.details);
-                        reject(error.message);
+
+                vscode.window.showInputBox({
```
This is quite some code repetition, is it possible to factor this out into a separate small function? (Showing the prompt, checking for the format, showing error message if violated, etc.)

phschaad commented on backend/run_dace.py:
```diff
@@ -53,7 +53,7 @@
 sys.path.append(path.abspath(path.dirname(__file__)))
 
-from dace_vscode import arith_ops, depth, transformations
+from dace_vscode import depth, transformations
```
Can't the `dace_vscode.arith_ops` module be removed entirely now?

phschaad commented on backend/dace_vscode/depth.py:
```diff
@@ -2,23 +2,45 @@
     get_exception_message,
 )
 
+def get_work(sdfg_json, assumptions):
```
This module should now probably be renamed to `work_depth` or something :-)

phschaad reviewed (changes requested):
Very nice, thank you :-) Small changes only

hodelcl commented:
I integrated all your comments:

- put the `InputBox` for the assumptions and the input checking in separate functions,
- deleted the `dace_vscode.arith_ops` module,
- renamed the `depth` module to `work_depth`.

phschaad reviewed (approved):
LGTM, thanks
